tle/cogs/contests.py

```diff
@@ -441,12 +441,43 @@ def _make_contest_embed_for_vc_ranklist(ranklist, vc_start_time=None, vc_end_tim
             embed.add_field(name='Tick tock', value=msg, inline=False)
         return embed
 
+    @staticmethod
+    def _filter_rated_only_contestant_data(handles, ranklist):
+
+        # Keep only rated contestants
+        rated_contestants = []
+        for handle in handles:
+            if ranklist.get_delta(handle) is not None:
+                rated_contestants.append(handle)
+
+        # fix the actual ranks for cases like Edu rounds where unofficial ranks are also included in official standings
+        current_rank = 0
+        last_rank = 0
+        last_score = (-1, -1)
+        for contestant in ranklist.standings:
+            handle = contestant.party.teamName or contestant.party.members[0].handle
+            try:
+                fk = ranklist.delta_by_handle[handle]
+                current_score = (contestant.points, contestant.penalty)
+                current_rank += 1
+                dict = ranklist.standing_by_id[handle]._asdict()
+                dict['rank'] = current_rank if current_score != last_score else last_rank
+                last_rank = dict['rank']
+                last_score = current_score
+                ranklist.standing_by_id[handle] = cf.make_from_dict(cf.RanklistRow, dict)
```

Comment:
I don't get how this filters out unrated contestants? ranklist entries are not changed for them, right? Shouldnt they be removed?

Reply:
The filtering out will happen by only keeping the rated contestant handles (notice rated_contestant list above which will only contain handles if there exist an delta for those -> hence rated)

This part of code is for fixing the correct ranks (as we noticed in the edu rounds that ranks are not actually the correct rank that are supposed for an official standings). 

Yes, it is true that thjs will only fix ranks for official participants and will not touch the other participants but this doesnt hurt as in later part of code those will be row removed as they wont have their handle in rated_contestant list

Reply:
Also an intermediate dict is made because RanklistRow is a named tuple which is immutable hence I have to create a new row to fix the rank